cli/pkg/edgeclient/client.go

```diff
@@ -96,13 +102,370 @@ func (c *Client) AddNewModelWithImage(
+			)
+			if err != nil {
+				return fmt.Errorf("failed to ensure artifact is in model registry: %w", err)
+			}
+		} else {
+			return fmt.Errorf("failed to ensure artifact is in model registry: %w", err)
+		}
+	} else if artifact == nil {
+		// Should never happen as we get an error if the artifact is not found, but just in case
+		return fmt.Errorf("failed to ensure artifact is in model registry: artifact not found")
+	}
+
+	return nil
 }
 
-func shortHash(s string) string {
-	hasher := sha256.New()
-	hasher.Write([]byte(s))
-	hash := hex.EncodeToString(hasher.Sum(nil))
-	return hash[:4]
+// CreatePipelineRun creates a tekton PipelineRun to build a model container image from a model version artifact.
+func (c *Client) CreatePipelineRun(
+	modelName string,
+	modelVersion string,
+	namespace string,
+	kubeConfig string,
+	parameters map[string]interface{},
+) (*PipelineRun, error) {
+	var s3SecretName string
+	var testDataConfigMapName string
+	var addDirWorkspace bool
+
+	if s3s, ok := parameters["s3SecretName"]; !ok {
+		return nil, fmt.Errorf("s3SecretName pipeline parameter is required")
+	} else if s3s, ok := s3s.(string); !ok {
+		return nil, fmt.Errorf("s3SecretName pipeline parameter must be a string")
+	} else {
+		s3SecretName = s3s
+	}
+
+	if tdc, ok := parameters["testDataConfigMapName"]; !ok {
+		return nil, fmt.Errorf("testDataConfigMapName pipeline parameter is required")
+	} else if tdc, ok := tdc.(string); !ok {
+		return nil, fmt.Errorf("testDataConfigMapName pipeline parameter must be a string")
+	} else {
+		testDataConfigMapName = tdc
+	}
+
+	if adw, ok := parameters["addDirWorkspace"]; !ok {
+		return nil, fmt.Errorf("addDirWorkspace pipeline parameter is required")
+	} else if adw, ok := adw.(string); !ok {
+		return nil, fmt.Errorf("addDirWorkspace pipeline parameter must be a boolean")
+	} else {
+		addDirWorkspace = adw == "true"
+	}
+
+	params, err := toTektonParams(modelName, modelVersion, parameters)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert parameters to tekton params: %w", err)
+	}
+
+	pipelineRun := &tektonv1.PipelineRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    namespace,
+			GenerateName: "aiedge-e2e-",
+			Labels: map[string]string{
+				"tekton.dev/pipeline": "aiedge-e2e",
+			},
+		},
+		Spec: tektonv1.PipelineRunSpec{
+			TaskRunTemplate: tektonv1.PipelineTaskRunTemplate{
+				ServiceAccountName: "pipeline",
+			},
+			Params: params,
+			PipelineRef: &tektonv1.PipelineRef{
+				Name: "aiedge-e2e",
+			},
+			Workspaces: []tektonv1.WorkspaceBinding{
+				{
+					Name: "build-workspace-pv",
+					VolumeClaimTemplate: &corev1.PersistentVolumeClaim{
+						Spec: corev1.PersistentVolumeClaimSpec{
+							AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+							Resources: corev1.VolumeResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceStorage: resource.MustParse("1Gi"),
```

Comment:
Should that 1Gi be configurable somehow? Not necessarily in this PR, but in general I think it would be useful.

Reply:
Probably we should do that in a separate task yes.